scipy/stats/tests/test_stats.py

```diff
@@ -255,19 +255,23 @@ def test_pROUNDROUND(self):
         r = y[0]
         assert_approx_equal(r,1.0)
 
-    def test_r_exactly_pos1(self):
+    def test_r_almost_exactly_pos1(self):
         a = arange(3.0)
-        b = a
-        r, prob = stats.pearsonr(a,b)
-        assert_equal(r, 1.0)
-        assert_equal(prob, 0.0)
+        r, prob = stats.pearsonr(a, a)
 
-    def test_r_exactly_neg1(self):
+        assert_allclose(r, 1.0, atol=1e-15)
+        # With n = len(a) = 3, the error in prob grows like the
+        # square root of the error in r.
+        assert_allclose(prob, 0.0, atol=np.sqrt(2*np.spacing(1.0)))
+
+    def test_r_almost_exactly_neg1(self):
```

Comment:
This could be condensed into the positive version of the test above with parametrization, but not worth the effort at this stage in my view.